plasmapy/diagnostics/_detector_stacks.py

```diff
@@ -0,0 +1,257 @@
+"""
+Objects representing stacks of film and/or filter layers for charged particle
+detectors.
+"""
+
+__all__ = [
+    "Stack",
+    "Layer",
+]
+
+import astropy.units as u
+import numpy as np
+
+from scipy.interpolate import interp1d
+from typing import List
+
+
+class Layer:
+    def __init__(
+        self,
+        thickness: u.m,
+        energy_axis: u.J,
+        stopping_power: u.MeV / u.cm,
+        active: bool = True,
+        name: str = "",
+    ):
+        r"""
+        A layer in a detector film stack. The layer could either be an active
+        layer (the actual film medium) or an inum_active layer (a filter or
+        inum_active part of the film, such as a substrate.)
+
+        Tabulated stopping powers for protons and electrons can be found in the
+        `NIST PSTAR database
+        <https://physics.nist.gov/PhysRefData/Star/Text/PSTAR.html>`_
+        and the
+        `NIST ESTAR database
+        <https://physics.nist.gov/PhysRefData/Star/Text/ESTAR.html>`_.
+
+        Parameters
+        ----------
+
+        thickness : `~astropy.units.Quantity`
+            The thickness of the layer, in units convertible to meters.
+
+        energy_axis : `~astropy.units.Quantity`
+            The energies corresponding to the stopping power array.
+
+        stopping_power : `~astropy.units.Quantity`
+            The stopping power in the material, multiplied by the material
+            mass density. The stopping power is tabulated in units of
+            MeV cm\ :sup:`^2` / g, so this variable has units of MeV/cm.
```

Comment:
This is tripping me up a bit since the term "stopping power" is being used for two different concepts, linear and mass stopping power, without clarification.  I think there's several options to proceed with, but I'm not sure with is the best option at the moment...

1. Keep the keyword argument `stopping_power`, but update the description to enforce that you mean the linear stopping power.
2. Keep the keyword argument `stopping_power`, but allow it to be either the linear or the mass stopping power.  This would also require an additional argument for the mass density of the film (It could possibly be optional 🤔).  However, the class attributes would need to be stored as `linear_stopping_power` and `mass_stopping_power`, instead of just `stopping_power`.

---

After writing this this, I think I'm favoring option 2.  Thoughts?